osu.Game/Graphics/UserInterfaceV2/ColourDisplay.cs

```diff
@@ -0,0 +1,105 @@
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Localisation;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Utils;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Graphics.UserInterfaceV2
+{
+    public class ColourDisplay : CompositeDrawable, IHasCurrentValue<Color4>
+    {
+        private readonly BindableWithCurrent<Color4> current = new BindableWithCurrent<Color4>();
+
+        private Box fill;
+        private OsuSpriteText colourHexCode;
+        private OsuSpriteText colourName;
+
+        public Bindable<Color4> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
+
+        private LocalisableString name;
+
+        public LocalisableString ColourName
+        {
+            get => name;
+            set
+            {
+                if (name == value)
+                    return;
+
+                name = value;
+
+                colourName.Text = name;
+            }
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            AutoSizeAxes = Axes.Y;
+            Width = 100;
+
+            InternalChild = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(0, 10),
+                Children = new Drawable[]
+                {
+                    new CircularContainer
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        Height = 100,
+                        Masking = true,
+                        Children = new Drawable[]
+                        {
+                            fill = new Box
+                            {
+                                RelativeSizeAxes = Axes.Both
+                            },
+                            colourHexCode = new OsuSpriteText
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Font = OsuFont.Default.With(size: 12)
+                            }
+                        }
+                    },
+                    colourName = new OsuSpriteText
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            current.BindValueChanged(_ => updateColour(), true);
+        }
+
+        private void updateColour()
+        {
+            fill.Colour = current.Value;
+            colourHexCode.Text = current.Value.ToHex();
```

Comment:
think it would be better using lower case, matching the design
```suggestion
            colourHexCode.Text = current.Value.ToHex().ToLowerInvariant();
```

Reply:
Tried this and am unconvinced. Old design used Exo, but with Torus the legibility of lowercase hex digits isn't great...

![2021-04-18-211721_677x209_scrot](https://user-images.githubusercontent.com/20418176/115157815-8cea5d80-a08b-11eb-87c9-1396a84a005e.png)

Reply:
Right, let's leave it at upper case then.